src/arp.rs

```diff
@@ -1,21 +1,299 @@
 use core::mem;
 
-/// ARP header, which is present after the Ethernet header.
+/// Represents an Address Resolution Protocol (ARP) header.
+///
+/// The ARP header is typically found after the Ethernet header and is used to
+/// map a network protocol address (like an IPv4 address) to a hardware
+/// address (like a MAC address).
 #[repr(C)]
 #[derive(Debug, Copy, Clone)]
 #[cfg_attr(feature = "serde", derive(::serde::Serialize, ::serde::Deserialize))]
 pub struct ArpHdr {
+    /// Hardware type (HTYPE): Specifies the network link protocol type.
+    /// E.g., Ethernet is 1.
     pub htype: [u8; 2],
+    /// Protocol type (PTYPE): Specifies the internetwork protocol for which
+    /// the ARP request is intended. For IPv4, this has the value 0x0800.
     pub ptype: [u8; 2],
+    /// Hardware address length (HLEN): Length in bytes of a hardware address.
+    /// Ethernet addresses size is 6.
     pub hlen: u8,
+    /// Protocol address length (PLEN): Length in bytes of a logical address.
+    /// IPv4 addresses size is 4.
     pub plen: u8,
+    /// Operation (OPER): Specifies the operation that the sender is performing:
+    /// 1 for request, 2 for reply.
     pub oper: [u8; 2],
+    /// Sender hardware address (SHA): The hardware address of the sender.
     pub sha: [u8; 6],
+    /// Sender protocol address (SPA): The protocol address of the sender.
     pub spa: [u8; 4],
+    /// Target hardware address (THA): The hardware address of the intended
+    /// receiver. This field is ignored in an ARP request.
     pub tha: [u8; 6],
+    /// Target protocol address (TPA): The protocol address of the intended
+    /// receiver.
     pub tpa: [u8; 4],
 }
 
+impl Default for ArpHdr {
+    /// Creates a new `ArpHdr` with all fields initialized to zero.
+    fn default() -> Self {
+        ArpHdr {
+            htype: [0; 2],
+            ptype: [0; 2],
+            hlen: 0,
+            plen: 0,
+            oper: [0; 2],
+            sha: [0; 6],
+            spa: [0; 4],
+            tha: [0; 6],
+            tpa: [0; 4],
+        }
+    }
+}
+
 impl ArpHdr {
+    /// The size of the ARP header in bytes.
     pub const LEN: usize = mem::size_of::<ArpHdr>();
+
+    /// Creates a new `ArpHdr` with all fields initialized to zero.
+    /// This is an alias for `ArpHdr::default()`.
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Returns the hardware type field.
+    #[inline]
+    pub fn htype(&self) -> [u8; 2] {
+        self.htype
+    }
+
+    /// Sets the hardware type field.
+    ///
+    /// # Arguments
+    ///
+    /// * `htype` - A 2-byte array representing the hardware type.
+    #[inline]
+    pub fn set_htype(&mut self, htype: [u8; 2]) {
+        self.htype = htype;
+    }
```

Comment:
Same here, let's use `u16` and convert it with in a performant way I proposed here:

```suggestion
    pub fn set_htype(&mut self, htype: u16) {
        self.htype = unsafe { *((&htype.swap_bytes() as *const _ as usize) as *const _) };
    }
```